catkin_tools_python/job.py

```diff
@@ -61,26 +61,23 @@ def fix_shebangs(logger, event_queue, pkg_dir, python_exec):
         if filename.endswith(('.py')):
             logger.out("Processing file ", filename)
             filepath = os.path.join(root, filename)
-            with open(filepath, 'r') as f:
-                try:
-                    contents = f.read()
-                except UnicodeDecodeError:
-                    logger.out("Unicode error caught skipping file : ", filepath)
-                    continue
+            with open(filepath, 'rb') as f:
+                contents = f.read()
 
+            new_shebang = ('#!%s' % python_exec).encode()
```

Comment:
Really, PYTHON_EXEC should probably also be a `bytes` so that this could just be 

    new_shebang = b'#!%s' % python_exec

However the source of that variable is either the `os.environ` dict or the `context.cmake_args` list, but of which have their values as strings, so this or `b'#!%s' % python_exec.encode()` is fine.